File: backend/src/main/java/com/vegnbio/api/modules/restaurant/entity/Restaurant.java
package com.vegnbio.api.modules.restaurant.entity;

import jakarta.persistence.*;
import lombok.*;

/**
 * Entité Restaurant représentant un restaurant VEG'N BIO
 * 
 * @author VegN-Bio Team
 */
@Entity @Table(name = "restaurants")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Restaurant {
  @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private String name;
  private String code;
  private String address;
  private String city;
  private String phone;
  private String email;
  
  // Nouvelles informations détaillées
  /** Indique si le Wi-Fi est disponible */
  private Boolean wifiAvailable;
  /** Nombre de salles de réunion disponibles */
  private Integer meetingRoomsCount;
  /** Capacité du restaurant en nombre de places */
  private Integer restaurantCapacity;
  /** Indique si une imprimante est disponible */
  private Boolean printerAvailable;
  /** Indique si les plateaux membres sont disponibles */
  private Boolean memberTrays;
  /** Indique si la livraison est disponible */
  private Boolean deliveryAvailable;
  /** Description des événements spéciaux */
  private String specialEvents;
  /** Horaires d'ouverture du lundi au jeudi */
  private String mondayThursdayHours;
  /** Horaires d'ouverture du vendredi */
  private String fridayHours;
  /** Horaires d'ouverture du samedi */
  private String saturdayHours;
  /** Horaires d'ouverture du dimanche */
  private String sundayHours;
}

File: vegn_bio_mobile/lib/models/restaurant.dart
class Restaurant {
  final int id;
  final String name;
  final String code;
  final String address;
  final String city;
  final String phone;
  final String email;
  
  // Nouvelles informations détaillées
  final bool? wifiAvailable;
  final int? meetingRoomsCount;
  final int? restaurantCapacity;
  final bool? printerAvailable;
  final bool? memberTrays;
  final bool? deliveryAvailable;
  final String? specialEvents;
  final String? mondayThursdayHours;
  final String? fridayHours;
  final String? saturdayHours;
  final String? sundayHours;

  Restaurant({
    required this.id,
    required this.name,
    required this.code,
    required this.address,
    required this.city,
    required this.phone,
    required this.email,
    this.wifiAvailable,
    this.meetingRoomsCount,
    this.restaurantCapacity,
    this.printerAvailable,
    this.memberTrays,
    this.deliveryAvailable,
    this.specialEvents,
    this.mondayThursdayHours,
    this.fridayHours,
    this.saturdayHours,
    this.sundayHours,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      code: json['code'] ?? '',
      address: json['address'] ?? '',
      city: json['city'] ?? '',
      phone: json['phone'] ?? '',
      email: json['email'] ?? '',
      wifiAvailable: json['wifiAvailable'],
      meetingRoomsCount: json['meetingRoomsCount'],
      restaurantCapacity: json['restaurantCapacity'],
      printerAvailable: json['printerAvailable'],
      memberTrays: json['memberTrays'],
      deliveryAvailable: json['deliveryAvailable'],
      specialEvents: json['specialEvents'],
      mondayThursdayHours: json['mondayThursdayHours'],
      fridayHours: json['fridayHours'],
      saturdayHours: json['saturdayHours'],
      sundayHours: json['sundayHours'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'code': code,
      'address': address,
      'city': city,
      'phone': phone,
      'email': email,
      'wifiAvailable': wifiAvailable,
      'meetingRoomsCount': meetingRoomsCount,
      'restaurantCapacity': restaurantCapacity,
      'printerAvailable': printerAvailable,
      'memberTrays': memberTrays,
      'deliveryAvailable': deliveryAvailable,
      'specialEvents': specialEvents,
      'mondayThursdayHours': mondayThursdayHours,
      'fridayHours': fridayHours,
      'saturdayHours': saturdayHours,
      'sundayHours': sundayHours,
    };
  }
}
